File: tests/runtime/many_arguments/host.py
import wasmtime
from many_arguments import ManyArguments, ManyArgumentsImports
from helpers import TestWasi

class MyImports:
    def many_arguments(self,
            a1: int,
            a2: int,
            a3: int,
            a4: int,
            a5: int,
            a6: int,
            a7: int,
            a8: int,
            a9: int,
            a10: int,
            a11: int,
            a12: int,
            a13: int,
            a14: int,
            a15: int,
            a16: int) -> None:
        assert(a1 == 1)
        assert(a2 == 2)
        assert(a3 == 3)
        assert(a4 == 4)
        assert(a5 == 5)
        assert(a6 == 6)
        assert(a7 == 7)
        assert(a8 == 8)
        assert(a9 == 9)
        assert(a10 == 10)
        assert(a11 == 11)
        assert(a12 == 12)
        assert(a13 == 13)
        assert(a14 == 14)
        assert(a15 == 15)
        assert(a16 == 16)


def run() -> None:
    store = wasmtime.Store()
    wasm = ManyArguments(store, ManyArgumentsImports(MyImports(), TestWasi()))

    wasm.many_arguments(store, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,14, 15, 16)

if __name__ == '__main__':
    run()

File: tests/runtime/many_arguments/host.rs
use anyhow::Result;

wit_bindgen_host_wasmtime_rust::generate!({
    import: "../../tests/runtime/many_arguments/imports.wit",
    default: "../../tests/runtime/many_arguments/exports.wit",
    name: "exports",
});

#[derive(Default)]
pub struct MyImports {}

impl imports::Imports for MyImports {
    fn many_arguments(
        &mut self,
        a1: u64,
        a2: u64,
        a3: u64,
        a4: u64,
        a5: u64,
        a6: u64,
        a7: u64,
        a8: u64,
        a9: u64,
        a10: u64,
        a11: u64,
        a12: u64,
        a13: u64,
        a14: u64,
        a15: u64,
        a16: u64,
    ) -> Result<()> {
        assert_eq!(a1, 1);
        assert_eq!(a2, 2);
        assert_eq!(a3, 3);
        assert_eq!(a4, 4);
        assert_eq!(a5, 5);
        assert_eq!(a6, 6);
        assert_eq!(a7, 7);
        assert_eq!(a8, 8);
        assert_eq!(a9, 9);
        assert_eq!(a10, 10);
        assert_eq!(a11, 11);
        assert_eq!(a12, 12);
        assert_eq!(a13, 13);
        assert_eq!(a14, 14);
        assert_eq!(a15, 15);
        assert_eq!(a16, 16);
        Ok(())
    }
}

fn run(wasm: &str) -> Result<()> {
    let (exports, mut store) = crate::instantiate(
        wasm,
        |linker| imports::add_to_linker(linker, |cx| -> &mut MyImports { &mut cx.imports }),
        |store, module, linker| Exports::instantiate(store, module, linker),
    )?;

    exports.many_arguments(
        &mut store, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16,
    )?;

    Ok(())
}
